### ISSUE TITLE
Menu with portal in Dialog

### ISSUE THREAD

**Issue by sbc640964** [tailwindlabs/headlessui#2208]
I'm using @headlessui/react:1.7.7 in chrome browser

You can see the bug here
https://codesandbox.io/s/headlessui-react-menu-example-forked-d9y765?file=/src/menu.js

The problem with Menu in a portal that exists in Dialog

I used with Menu in a portal like this https://github.com/tailwindlabs/headlessui/blob/main/packages/playground-react/pages/menu/menu-with-popper.tsx

I placed it inside a dialog

The keyboard accessibility does not work well, you will see that the keyboard moves between the last and the first item only

Apparently the problem is this section of the documentation
![image](https://user-images.githubusercontent.com/46646044/214517656-ee507bb3-1297-4276-b0db-f090c9be28a7.png)
